akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardingQueries.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.sharding
+
+import scala.reflect.ClassTag
+
+import akka.annotation.InternalApi
+
+/** INTERNAL API */
+@InternalApi
+private[sharding] object ShardingQueries {
+
+  /**
+   * INTERNAL API
+   * The result of a group query and metadata.
+   *
+   * @param failed the queries to shards that failed within the configured timeout. This
+   *                     could be indicative of several states, e.g. still in initialization,
+   *                     restart, heavily loaded and busy, where returning zero entities is
+   *                     not indicative of the reason
+   * @param responses the responses received from the query
+   * @param total the total number of shards tracked versus a possible subset
+   * @param queried the number of shards queried, which could equal the total or be a
+   *                subset if this was a retry of those that timed out
+   * @tparam A
+   * @tparam B
+   */
+  final case class ShardsQueryResult[A, B](failed: Set[A], responses: Seq[B], total: Int, queried: Int) {
+
+    /** Returns true if there was anything to query. */
+    def nonEmpty: Boolean = total > 0 && queried > 0
```

Comment:
Called 2x, hence the def vs val or nothing

Reply:
Well on this now we can do a val yes. But to not have it does mean even with It's `private[sharding]` we'd be calling `total > 0 && queried > 0` more than once in one function which I don't like. Updating to val though, that is correct now. Thanks.

Reply:
pushed.